File: my_app/app/src/main/java/com/example/my_app/cook/navigation/OrderAdapter.java
package com.example.my_app.cook.navigation;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;
import com.squareup.picasso.Picasso;
import com.example.my_app.R;

import java.util.ArrayList;

public class OrderAdapter extends RecyclerView.Adapter<OrderAdapter.OrderViewHolder> {

    private ArrayList<OrderItem> mOrderList;
    public static class OrderViewHolder extends RecyclerView.ViewHolder {
        public ImageView mImageView;
        public TextView mTextView1;
        public TextView mTextView2;
        public TextView mTextView3;

        public OrderViewHolder(@NonNull View itemView) {
            super(itemView);
            mImageView = itemView.findViewById(R.id.imageView);
            mTextView1 = itemView.findViewById(R.id.textView);
            mTextView2 = itemView.findViewById(R.id.textView1);
            mTextView3 = itemView.findViewById(R.id.textView2);
        }
    }

    public OrderAdapter(ArrayList<OrderItem> orderList) {
        mOrderList = orderList;
    }
    @NonNull
    @Override
    public OrderViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.order_item, parent, false);
        OrderViewHolder ovh = new OrderViewHolder(view);
        return ovh;
    }

    @Override
    public void onBindViewHolder(@NonNull OrderViewHolder holder, int position) {
        OrderItem currentItem = mOrderList.get(position);

        if (currentItem.getImageUrl() == null) {
            holder.mImageView.setImageResource(R.drawable.common_google_signin_btn_icon_disabled);
        }
        else if (currentItem.getImageUrl().trim().equals("")) {
            holder.mImageView.setImageResource(R.drawable.common_google_signin_btn_icon_disabled);
        }
        else {
            Picasso.with(holder.mImageView.getContext())
                    .load(currentItem.getImageUrl())
                    .resize(80, 80)
                    .centerCrop()
                    .into(holder.mImageView);
        }
        holder.mTextView1.setText("Order: " + currentItem.getOrder());
        holder.mTextView2.setText("Customer: " + currentItem.getCustomer());
        holder.mTextView3.setText("Vendor: " + currentItem.getVendor());
    }

    @Override
    public int getItemCount() {
        return mOrderList.size();
    }
}

File: BK_FoodCourt/app/src/main/java/com/example/bk_foodcourt/order/orderList/OrderAdapter.kt
package com.example.bk_foodcourt.order.orderList

import android.util.Log
import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.example.bk_foodcourt.databinding.OrderItemBinding
import com.example.bk_foodcourt.order.Order
import com.firebase.ui.firestore.FirestoreRecyclerAdapter
import com.firebase.ui.firestore.FirestoreRecyclerOptions

class OrderAdapter(options: FirestoreRecyclerOptions<Order>, val viewModel: OrderViewModel):
    FirestoreRecyclerAdapter<Order, OrderAdapter.OrderViewHolder>(options) {

    class OrderViewHolder private constructor(private val binding: OrderItemBinding): RecyclerView.ViewHolder(binding.root) {

        fun bind(item: Order, viewModel: OrderViewModel) {
            binding.order = item
            binding.viewModel = viewModel
        }

        companion object {
            fun from(parent: ViewGroup): OrderViewHolder {
                val layoutInflater = LayoutInflater.from(parent.context)
                val binding = OrderItemBinding.inflate(layoutInflater, parent, false)

                return OrderViewHolder(binding)
            }
        }
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): OrderViewHolder {
        Log.d("OrderAdapter", "Create order view holder")
        return OrderViewHolder.from(parent)
    }

    override fun onBindViewHolder(holder: OrderViewHolder, position: Int, item: Order) {
        item.id = snapshots.getSnapshot(position).id
        holder.bind(item, viewModel)
    }
}